cmd/sts-handlers.go

```diff
@@ -365,54 +367,11 @@ func (sts *stsAPIHandlers) AssumeRoleWithSSO(w http.ResponseWriter, r *http.Requ
 		return
 	}
 
-	// REQUIRED. Audience(s) that this ID Token is intended for.
-	// It MUST contain the OAuth 2.0 client_id of the Relying Party
-	// as an audience value. It MAY also contain identifiers for
-	// other audiences. In the general case, the aud value is an
-	// array of case sensitive strings. In the common special case
-	// when there is one audience, the aud value MAY be a single
-	// case sensitive
-	audValues, ok := iampolicy.GetValuesFromClaims(m, audClaim)
-	if !ok {
-		writeSTSErrorResponse(ctx, w, true, ErrSTSInvalidParameterValue,
-			errors.New("STS JWT Token has `aud` claim invalid, `aud` must match configured OpenID Client ID"))
-		return
-	}
-	if !audValues.Contains(globalOpenIDConfig.ClientID) {
-		// if audience claims is missing, look for "azp" claims.
-		// OPTIONAL. Authorized party - the party to which the ID
-		// Token was issued. If present, it MUST contain the OAuth
-		// 2.0 Client ID of this party. This Claim is only needed
-		// when the ID Token has a single audience value and that
-		// audience is different than the authorized party. It MAY
-		// be included even when the authorized party is the same
-		// as the sole audience. The azp value is a case sensitive
-		// string containing a StringOrURI value
-		azpValues, ok := iampolicy.GetValuesFromClaims(m, azpClaim)
-		if !ok {
-			writeSTSErrorResponse(ctx, w, true, ErrSTSInvalidParameterValue,
-				errors.New("STS JWT Token has `aud` claim invalid, `aud` must match configured OpenID Client ID"))
-			return
-		}
-		if !azpValues.Contains(globalOpenIDConfig.ClientID) {
-			writeSTSErrorResponse(ctx, w, true, ErrSTSInvalidParameterValue,
-				errors.New("STS JWT Token has `azp` claim invalid, `azp` must match configured OpenID Client ID"))
-			return
-		}
-	}
-
```

Comment:
Not sure why this is removed @donatello - not able to find a good explanation here.